src/sync/remote_mongo_collection.cpp

```diff
@@ -407,24 +261,230 @@ void MongoCollection::find_one_and_replace(const bson::BsonDocument& filter_bson
 void MongoCollection::find_one_and_delete(const bson::BsonDocument& filter_bson,
                                           MongoCollection::FindOneAndModifyOptions options,
                                           std::function<void(util::Optional<bson::BsonDocument>, util::Optional<AppError>)> completion_block)
+{
+    find_one_and_delete_bson(filter_bson, options, handle_document_response(completion_block));
+}
+
+void MongoCollection::find_one_and_delete(const bson::BsonDocument& filter_bson,
+                                          std::function<void(util::Optional<bson::BsonDocument>, util::Optional<AppError>)> completion_block)
+{
+    find_one_and_delete(filter_bson, {}, completion_block);
+}
+
+void MongoCollection::find_bson(const bson::BsonDocument& filter_bson,
+                                FindOptions options,
+                                std::function<void(util::Optional<AppError>, util::Optional<bson::Bson>)> completion_block)
+{
+    try {
```

Comment:
If you want to just wrap the entire function in a try catch block, you can make this a little cleaner by doing:
```
void FunctionDeclaration(foo bar, bizz buzz) try
{
   {function body}
} catch(const std::exception& e) {
   { catch block 1 }
} catch(const OtherException& e) {
   { catch block 2 }
}
```

See https://en.cppreference.com/w/cpp/language/function-try-block for more details.

Reply:
That's some overly fancy syntax here 😄